lib/index.js

```diff
@@ -154,6 +154,15 @@ internals.handler = function (route, handlerOptions) {
                 return reply(err);
             };
 
+            // Encoding request path
+            const _encodeURL = (path) => {
+                if (decodeURI(path) === path){
+                    path = encodeURI(path);
+                }       
+                return path;
+            };
+            uri = _encodeURL(uri);
```

Comment:
does uri contains ONLY path or it is a full URL ?
If it is a full URL change the variable name to url 
If NOT then pass ONLY the path 
Maybe names are just a bit confusing here

Reply:
It's full path, I will change it to url